### PR TITLE
Clockcult Patch

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Added sound effects and tweaked messages for being ice-cubed by freon gas.
add: Added the Stunning Flare scripture, which allows Ratvar's servants to conjure a blinding flare that stuns and mutes in a melee attack. It is a Driver that costs 2 belligerent eyes.
add: Replaced Sentinel's Compromise with Sentinel's Endowment, which heals 60 brute and burn damage over 1 minute.
add: The clockwork slab now has a built-in personality.
tweak: Redid clockwork slab buttons. An actual interface is coming soon(tm).
tweak: Judicial Visor is now a Script and has had its component requirements changed accordingly.
tweak: Edited some descriptions and messages for clockcult scripture and items.
tweak: Completely rewrote Recollection to be less wordy and more of an actual guide.
tweak: Clockwork helmets and gauntlets are less protective against melee attacks.
/:cl:

New sound effects can be found here: [freezing](https://dl.dropboxusercontent.com/u/116194515/ice_form.ogg) and [breaking out](https://dl.dropboxusercontent.com/u/116194515/ice_shatter.ogg)
Numerical values:
- Stunning Flare stuns for 10 seconds and silences for 5; the same as a stun talisman. Additionally, the victim is confused for 15 seconds
  - Nar-Sie's cultists will be confused for 30 seconds, cultslur for 15 seconds, and take 15 fire damage.
- Clockwork helmet and gauntlet brute protection reduced to 50 (down from 80). Chestplate remains unaffected.

WIP; more to come. Feel free to discuss what's already here.

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_items.dm [hunk 1]
```diff
@@ -253,40 +253,51 @@
...
+		more static. That means that, instead of staying on the move, we make fortified strongholds and set up our operations there. Nezbere, Ratvar's engineer, has designed a lot of structures, \
+		so we need somewhere to put them. The most important of these is the <b>tinkerer's cache</b>, which lets you store components for later use. You can offload my (the slab) components into \
+		one by hitting it with me. Caches are so good because <i>everyone</i> can draw from the components inside, all caches are linked, and the components can be used to power scripture! It's \
+		a wonderful thing, teamwork. Anyway, the cache and other structures have something called <b>CV</b> - that stands for Construction Value. When you see CV mentioned in a tier unlock, we \
+		need that much <i>combined</i> CV to unlock it.<br><br>\
 		\
-		A complete list of scripture, its effects, and its requirements can be found below.<br>\
+		Ol gur Nex, listen to me ramble. You probably just want to know the scripture, so let me tell you. I can't teach you any more, but feel free to ask your fellow servants for help. Anyway, \
+		here's your scripture...<br>\
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items.dm]: this is multiple lines LONGER than it was before

File: code/game/gamemodes/clock_cult/clock_items.dm [hunk 2]
```diff
@@ -540,21 +551,65 @@
...
+
 /obj/item/clothing/head/helmet/clockwork //Clockwork armor: High melee protection but weak to lasers
 	name = "clockwork helmet"
 	desc = "A heavy helmet made of brass."
 	icon = 'icons/obj/clothing/clockwork_garb.dmi'
 	icon_state = "clockwork_helmet"
 	w_class = 3
 	resistance_flags = FIRE_PROOF | ACID_PROOF
-	armor = list(melee = 80, bullet = 50, laser = -15, energy = 0, bomb = 35, bio = 0, rad = 0, fire = 100, acid = 100)
+	armor = list(melee = 50, bullet = 50, laser = -15, energy = 0, bomb = 35, bio = 0, rad = 0, fire = 100, acid = 100)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items.dm]: I'm not sure why you're nerfing the melee armor of this, given the laser weakness it has...

File: code/game/gamemodes/clock_cult/clock_ratvar.dm [hunk 3]
```diff
@@ -187,7 +187,7 @@
 /obj/structure/destructible/clockwork/massive/ratvar
 	name = "Ratvar, the Clockwork Justiciar"
 	desc = "<span class='userdanger'>What is what is what are what real what is all a lie all a lie it's all a lie why how can what is</span>"
-	clockwork_desc = "<span class='large_brass'><b><i>Ratvar, the Clockwork Justiciar, your master eternal.</i></b></span>"
+	clockwork_desc = "It's Ratvar. You kind of expected more fanfare."
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_ratvar.dm]: I'm not really sure why you're changing this

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 4]
```diff
@@ -287,23 +295,8 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
...
-	L.adjustFireLoss(-burndamage)
-	var/healseverity = max(round(totaldamage*0.05, 1), 1) //shows the general severity of the damage you just healed, 1 glow per 20
-	var/targetturf = get_turf(L)
-	for(var/i in 1 to healseverity)
-		PoolOrNew(/obj/effect/overlay/temp/heal, list(targetturf, "#1E8CE1"))
 	invoker << "<span class='brass'>You bathe [L] in Inath-neq's power!</span>"
-	L.visible_message("<span class='warning'>A blue light washes over [L], mending [L.p_their()] bruises and burns!</span>", \
-	"<span class='heavy_brass'>You feel Inath-neq's power healing your wounds, but a deep nausea overcomes you!</span>")
-	playsound(targetturf, 'sound/magic/Staff_Healing.ogg', 50, 1)
+	L.apply_status_effect(STATUS_EFFECT_SENTINELS_BOON)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: I'd like to note that this is much, much weaker than the Compromise, given that it's 1/second for two damage types.

File: code/modules/mob/living/living_defense.dm [hunk 5]
```diff
@@ -323,6 +323,8 @@
 
 
 /mob/living/ratvar_act()
+	if(is_servant_of_ratvar(src))
```
Comment (ChangelingRain) [inline code/modules/mob/living/living_defense.dm]: Just change the below check to `if(!is_servant_of_ratvar(src) && !add_servant_of_ratvar(src))`

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 6]
```diff
@@ -687,6 +680,25 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
 
 
 
+/datum/clockwork_scripture/create_object/judicial_visor //Judicial Visor: Creates a judicial visor.
+	descname = "Delayed Area Stun Glasses"
+	name = "Judicial Visor"
+	desc = "Forms a visor that, when worn, will grant the ability to form a flame in your hand that can be activated at an area to smite it, stunning, muting, and damaging the nonfaithful. \
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: This desc is outdated, as it no longer requires a hand.

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 7]
```diff
@@ -221,23 +221,31 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
...
 
+/datum/clockwork_scripture/stunning_flare/check_special_requirements()
+	if(!invoker.get_empty_held_index_for_side("l") && !invoker.get_empty_held_index_for_side("r"))
+		invoker << "<span class='warning'>You need a free hand to hold the flare!</span>"
+		return 0
+	return 1
+
+/datum/clockwork_scripture/stunning_flare/scripture_effects()
+	invoker <<  "<span class='brass'>You copy a piece of replicant alloy and command it into a new slab.</span>" //No visible message, for stealth purposes
+	var/obj/item/weapon/ratvars_light/L = new(get_turf(invoker))
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: I'd kind of prefer if this used something similar to what the judicial visor uses, with a ranged ability and a cancel button, and delaying the scripture returning until they either use the flare, cancel, or become unable to use it

File: code/game/gamemodes/clock_cult/clock_items.dm [hunk 8]
```diff
@@ -540,21 +551,65 @@
...
+	force = 5 //Also serves as a weak melee weapon!
+	damtype = BURN
+	hitsound = 'sound/weapons/sear.ogg'
+	attack_verb = list("scorched", "seared", "burnt")
+
+/obj/item/weapon/ratvars_light/examine(mob/user)
+	..()
+	user << "<span class='brass'>Attack a human to stun and blind them.</span>"
+
+/obj/item/weapon/ratvars_light/afterattack(atom/target, mob/living/user, flag, params)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items.dm]: This is not melee. This is, in fact, ridiculously ranged

**Review by ChangelingRain — COMMENTED** [review]
A: You're going to conflict forever
B: as much as clockcult has issues I really, really don't like giving them what is basically a stunpaper

File: code/game/gamemodes/clock_cult/clock_items.dm [hunk 9]
```diff
@@ -169,40 +168,41 @@
...
 		playsound(get_turf(user), 'sound/weapons/sear.ogg', 50, 1)
 		user.drop_item()
 		user.emote("scream")
 		user.apply_damage(5, BURN, "l_arm")
 		user.apply_damage(5, BURN, "r_arm")
 		return 0
 	if(!is_servant_of_ratvar(user))
-		user << "<span class='warning'>The information on [src]'s display shifts rapidly. After a moment, your head begins to pound, and you tear your eyes away.</span>"
+		user << "<span class='heavy_brass'>\"[text2ratvar("Sorry, buddy, but unless you rock the Engine roll, this ain't happening.")]\"</span>"
+		user << "<span class='warning'>You realize you've been staring at [src]'s clock in a hypnosis. You quickly loook away.</span>"
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items.dm]: loook

**Comment by Xhuis** [comment]
Right, conflicts fixed. Responding to a few comments now.

File: code/modules/mob/living/living_defense.dm (hunk 5 above)
Comment (Xhuis) [inline code/modules/mob/living/living_defense.dm]: Sure, I'll try that.

File: code/game/gamemodes/clock_cult/clock_scripture.dm (hunk 6 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: I'll replace that.

File: code/game/gamemodes/clock_cult/clock_scripture.dm (hunk 4 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: That's partially the intention. Like a lot of Ratvar's scripture, this was very situational but beat every other option in that situation. I felt that a HoT was, although weaker overall, more useful in more situations; for instance, you could pre-HoT before a fight, get someone out of critical condition, or just recover from being shocked.

File: code/game/gamemodes/clock_cult/clock_ratvar.dm (hunk 3 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_ratvar.dm]: I was told the other one was too edgy.

File: code/game/gamemodes/clock_cult/clock_items.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items.dm]: I'm aware, but it's also more informative. I feel like, in the long run, it'll help more.

File: code/game/gamemodes/clock_cult/clock_items.dm (hunk 8 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items.dm]: Ooh, I forgot to put an adjacency check. Good eye!

File: code/game/gamemodes/clock_cult/clock_ratvar.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_ratvar.dm]: This isn't edgy, the standard desc is!

File: code/game/gamemodes/clock_cult/clock_items.dm (hunk 2 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items.dm]: I feel like the small laser weakness isn't enough to offput it. You only take 3 more damage per laser from this, and the other protections it offers are insane compared to that. Either I could increase the laser damage taken or reduce the brute protection. That, or something else, if you have a better idea than I do.

File: code/game/gamemodes/clock_cult/clock_items.dm (hunk 9 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_items.dm]: That's actually intentional ;D

**Comment by ChangelingRain** [comment]
The tried and true option is fucking horrible to deal with and this is actually worse, because there's no tell if you have it in your hand like a stunpaper, which is bright fucking red.

**Comment by Xhuis** [comment]
I wasn't entirely sure how it being tied to the slab should work. I suppose I could have your hand on fire, but I would much rather have a more elegant solution. Any ideas?

File: code/game/gamemodes/clock_cult/clock_scripture.dm (hunk 4 above)
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: I don't think 'situational but super goddamn good in that situation' is bad, though the interface complicates this somewhat.

File: code/game/gamemodes/clock_cult/clock_items.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items.dm]: Increase the laser weakness, maybe, but reducing the melee armor makes the armor way less useful.

File: code/game/gamemodes/clock_cult/clock_ratvar.dm (hunk 3 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_ratvar.dm]: I always thought that it was just your mind trying to comprehend the fact that a literal god is in front of you. I'll update them both.

**Comment by ChangelingRain** [comment]
`<Xhuis> No joke don't merge my PR, it's a wip`

will oranges listen find out tomorrow

**Comment by Xhuis** [comment]
#21177 makes this too troublesome and conflicting to work on right now. I'll remake this once that's in.

**Pull request closed without merging**